src/Uri.php

```diff
@@ -333,17 +335,13 @@ public function withHost($host)
      */
     public function withPort($port)
     {
-        if (! is_numeric($port) && $port !== null) {
+        if (! is_int($port) && $port !== null) {
```

Comment:
This is problematic, as the port value _may be derived from a string_. This is why the check was originally for `is_numeric`, and we cast to `int`. While PHP's `parse_url()` does the right thing, and casts the discovered port to an integer, the value might come from another location:

- from `$_SERVER`, in which case it's a string
- by splitting the authority portion of the URI, in which case it's a string
- by manually parsing the URI in order to provide a more robust solution than `parse_url()`, in which case casting would be up to the parser

Requiring an `int` at this point is a BC break.